.mergify.yml

```diff
@@ -0,0 +1,181 @@
+# Inital state (needs: review) is done by internal reviewer bot
+
+# Notes: conditions do not check if label is already there if we add or remove later, this is to limit Github API access to
+# minimum. If a label is already there and we add it again, nothing should happen (we do 1 less access to Github).
+# For instance for needs: work if CI failed, we just add a label as action, don't check if needs: works is already there
+
+pull_request_rules:
+  # From needs: review to needs: work - reviewers request
+  - name: "label needs: work when changes were requested"
+    conditions:
+      - base~=feature-mergify
+      # Reviewers
+      - "#changes-requested-reviews-by>0"
+    actions:
+      label:
+        add: 
+          - "needs: work"
+        remove: 
+          - "needs: review"
+
+  # From needs: review to needs: work - CI failure
+  - name: "label needs: work when travis-ci failed"
+    conditions:
+      - base~=feature-mergify
+      # Travis failing
+      - status-failure~=continuous-integration/travis-ci/pr
+    actions:
+      label:
+        add: 
+          - "needs: work"
+        remove: 
+          - "needs: review"
+
+  # From needs: review to needs: work - CI failure
+  - name: "label needs: work when Jenkins CI failed - pr head"
+    conditions:
+      - base~=feature-mergify
+      # Jenkins CI failing
+      - status-failure~=continuous-integration/jenkins/pr-head
+    actions:
+      label:
+        add: 
+          - "needs: work"
+        remove: 
+          - "needs: review"
+
+  # From needs: review to needs: work - CI failure
+  - name: "label needs: work when Jenkins CI failed - any of the pipeline"
+    conditions:
+      - base~=feature-mergify
+      # Jenkins CI failing - any of the pipeline
+      - status-failure~=^jenkins-ci
+    actions:
+      label:
+        add: 
+          - "needs: work"
+        remove: 
+          - "needs: review"
+
+  # From needs: review to needs: CI
+  - name: "label needs: CI when at least one reviewers approval"
+    conditions:
+      - base~=feature-mergify
+      # Labels
+      - "label!=needs: work"
+      - "label!=needs: preceding PR"
+
+      # Reviewers
+      - "#changes-requested-reviews-by=0"
+      - "#approved-reviews-by>=1"
+
+      # CI green policy, at least Travis should be green
+      - status-success~=continuous-integration/travis-ci/pr
+      # new CI needs to be done
+      - status-neutral~=continuous-integration/jenkins/pr-head
+    actions:
+      label:
+        add: 
+          - "needs: CI"
+        remove: 
+          - "needs: review"
+
+  # Conflict in the PR - needs: work and a comment to notify a user
+  - name: "label needs: work when there is a conflict"
+    conditions:
+      - base~=feature-mergify
+      - conflict
+    actions:
+      label:
+        add:
+          - "needs: work"
+        remove: 
+          - "needs: review"
+      comment:
+        message: This PR cannot be merged due to conflicts. Please rebase to resolve them.
```

Comment:
Could we also get a conflict when in the 'needs: ci' state ?

Reply:
I see, we should remove as well needs: CI, will fix

Reply:
fixed